console/v1alpha1/types_console_plugin.go

```diff
@@ -27,16 +27,49 @@ type ConsolePluginSpec struct {
 	// service serving certificate. The console backend will proxy the
 	// plugins assets from the Service using the service CA bundle.
 	Service ConsolePluginService `json:"service"`
+	// proxy is a list of services that the plugin needs to connect to.
+	Proxy ConsolePluginProxy `json:"proxy"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// services is list of Kubernetes Service to which the plugin
+	// will connect to, using a deployment with an HTTP server.
+	// The Service must use HTTPS and service serving certificate.
+	// The console backend will proxy the communication between
+	// plugin and the specified Service.
```

Comment:
We should document the path that plugins can use to make requests as described in the enhancement.